<% 
let _ttl;
try { 
    _ttl = ttl
}
catch (e) {
    _ttl = 0
}
finally {
    ttl = _ttl
}
let _user;
try {
    _user = user
}
catch (e) {
    _user = null
}
finally {
    user = _user
}
%>
<%
    const error = ({
        401: {
            title: 'Unauthorized',
            message: 'You are not authorized to access this page',
            linksText: user 
                ? '<a href="/">Go to the main page</a>'
                : '<a href="/login">Login</a> or <a href="signup">Sign up</a>',
        },
        403: {
            title: 'Forbidden',
            message: 'You are not authorized to access this page',
            linksText: user 
                ? '<a href="/">Go to the main page</a>'
                : '<a href="/login">Login</a> or <a href="signup">Sign up</a>',
        },
        404: {
            title: 'Not Found',
            message: 'The page you are looking for does not exist',
            linksText: '<a href="/">Go to the main page</a>',
        },
        429: {
            title: 'Too Many Requests',
            message: 'You exceeded the maximum number of requests',
            linksText: `Try again after <strong>${ttl}</strong> seconds`,
        }
    })[code]
%>

<!DOCTYPE html>
<html lang="en">
<head>
    <%- include('partial/head') %>
    <title><%= error.title %></title>
</head>
<body>
    
    <div class="container">
        <h1><code><%= code %></code> <%= error.title %></h1>
        <p><%= error.message %></p>
        <%- error.linksText %>
    </div>

</body>
</html>